### PR TITLE
Add the NREL-5MW turbine controller written in C

### PR DESCRIPTION
**Pull request by faisal-bhuiyan**
Implements #155

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@
 ############################ BASE ######################################
 
 cmake_minimum_required (VERSION 3.20 FATAL_ERROR)
-project(OpenTurbine LANGUAGES CXX)
+project(OpenTurbine LANGUAGES C CXX)
```
Comment (ddement) [inline CMakeLists.txt:5]: Why are you writing this in C?  I see no reason to not do this in C++ like the rest of our code.

Comment (faisal-bhuiyan) [reply CMakeLists.txt:5]: This was already written in C, please see the issue description.

File: tests/unit_tests/restruct_poc/test_controller.cpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    // Test data generated using the following regression test from the
+    // OpenFAST/r-test repository:
+    // https://github.com/OpenFAST/r-test/tree/main/glue-codes/openfast/5MW_Land_DLL_WTurb
+    // at time = 0.0s
+
+    // Use dylib to load the dynamic library and get access to the controller functions
+    util::dylib lib("./", "DISCON");
+    auto DISCON = lib.get_function<void(float*, int&, char*, char*, char*)>("DISCON");
+
+    float avrSWAP[81] = {0.};
```
Comment (deslaughter) [inline tests/unit_tests/restruct_poc/test_controller.cpp:45]: After avrSWAP is initialized to zero, I recommend doing `SwapStruct* swap = reinterpret_cast<SwapStruct*>(avrSWAP);` and setting the values through the struct which will do a better job of documenting what is being set than the comments.

**Review by deslaughter — COMMENTED** [review]
Looks pretty good, just a couple of comments.

File: src/utilities/controllers/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+target_sources(${oturb_lib_name}
```
[the remark this answers is not in the document]
Comment (faisal-bhuiyan) [reply src/utilities/controllers/CMakeLists.txt:1]: Good catch, I was experimenting with a bunch of stuff to fix the name mangling issue. We do not need to include `discon.cpp` to openturbine.

**Review by deslaughter — APPROVED** [review]

**Pull request merged**